src/java_bytecode/java_bytecode_parse_tree.h

```diff
@@ -80,6 +82,12 @@ class java_bytecode_parse_treet
   {
   public:
     irep_idt base_name;
+
+    bool is_constructor() const
+    {
+      return base_name=="<init>";
```

Comment:
Nooo -  this re-duplicates the check that `<init>` is the name of constructors that I've painstakingly removed once! Either leave as is of remove the `is_constructor` in `java_bytecode_convert_method`

Reply:
I think that `is_constructor` in `java_bytecode_convert_method` can be entirely removed. I'd replace the call in java_bytecode_convert_method.cpp:381 to use this new method. The other two uses java_bytecode_convert_method.cpp:575 and 1327 can be replaced by `type.get_is_constructor()`.